bindings/src/Capgemini.PowerApps.SpecFlowBindings/TestDriver.cs

```diff
@@ -46,8 +46,8 @@ public void InjectOnPage(string authToken)
             if (!string.IsNullOrEmpty(authToken))
             {
                 scriptBuilder.AppendLine(
-                    $@"var appUserRecordRepository = new {LibraryNamespace}.AuthenticatedRecordRepository(metadataRepository, '{authToken}');
-                       var dataManager = new {LibraryNamespace}.DataManager(recordRepository, deepInsertService, [new {LibraryNamespace}.FakerPreprocessor()], appUserRecordRepository);");
+                    $@"window.appUserRecordRepository = new {LibraryNamespace}.AuthenticatedRecordRepository(metadataRepository, '{authToken}');
```

Comment:
I feel like using `window` here rather than `top` could have unintended consequences. Depending on which frame is active, what is created in the `window` object could be lost during navigation. The only `window` that remains constant is `top`. 

`TestDriverReference` is already `top.driver`, so I think anywhere in `InjectOnPage` where we've got `window` or `var` we should update to `top`.

```suggestion
                    $@"top.appUserRecordRepository = new {LibraryNamespace}.AuthenticatedRecordRepository(metadataRepository, '{authToken}');
```

Reply:
We did try this but let's try again

Reply:
To revert

```suggestion
                    $@"var appUserRecordRepository = new {LibraryNamespace}.AuthenticatedRecordRepository(metadataRepository, '{authToken}');
```